lib/cdo/poste.rb

```diff
@@ -111,23 +112,22 @@ def self.unsubscribe(email, hashed_email, params={})
 
   class Template
     def initialize(path)
-      body = IO.read(path)
+      @path = path
 
       @engine = {
         '.haml' => TextRender::HamlEngine,
         '.html' => TextRender::ErbEngine,
         '.md' => TextRender::MarkdownEngine,
       }[File.extname(path).downcase]
 
-      if match = body.match(/^---\s*\n(?<header>.*?\n?)^(---\s*$\n?)(?<html>\s*\n.*?\n?)^(---\s*$\n?)(?<text>\s*\n.*?\n?\z)/m)
-        @header = match[:header].strip
-        @html = match[:html].strip
-        @text = match[:text].strip
-      elsif match = body.match(/^---\s*\n(?<header>.*?\n?)^(---\s*$\n?)(?<html>\s*\n.*?\n?\z)/m)
-        @header = match[:header].strip
-        @html = match[:html].strip
-      else
-        @html = body.strip
+      @header, @html, @text = parse_template(IO.read(path))
+
+      # Temporarily also load in a new 'actionview' template and render it
+      # alongside the given one; we will use this to help build confidence that
+      # switching our templates over to actionview can be done without changing
+      # the resulting rendered HTML
+      if File.exists?(path + ".actionview")
+        @actionview_header, @actionview_html, @actionview_text = parse_template(IO.read(path + ".actionview"))
       end
```

Comment:
Should there be an else-case here that also notifies Honeybadger, in case someone adds a new email template between now and the cutover?

Reply:
ooh, there definitely should